askama_derive/src/generator.rs

```diff
@@ -1921,6 +1921,18 @@ impl<'a, K: Eq + hash::Hash, V> Default for MapChain<'a, K, V> {
     }
 }
 
+/// Returns `true` if an element of this `Expr` is a function/method call.
+fn has_call(expr: &Expr<'_>) -> bool {
+    use Expr::*;
+    match expr {
+        Filter(_, _) | Call(_, _) | RustMacro(_, _) => true,
```

Comment:
For each of these arms, can you briefly summarize your thinking on why you choose this outcome?

Reply:
In a comment in the code or here? Let's start by only here and you'll tell me if you want it as a comment in the code:

For `Filter`, it basically calls a function (the filter), potentially on multiple levels (each filter becomes an argument to the next one). So with the reference, it'd be the same as doing `let x = &filter(blabla);`. Hence a reference to a temporary value.
For `Call`, it's a function call, so same as `Filter`.
For `RustMacro`, we have no idea what will actually end up being generated so better not try to wrap it in a reference either because the last macro element could very much be a function/method call (then back to argument for `Filter`).

Does my logic make sense?